### ISSUE TITLE
Non-configurable quadBuffer size

### ISSUE THREAD

**Issue by elsassph** [lightning-js/renderer#286]
LightningJS 2 allowed to configure the internal buffer size - its default value was 2Mb, while it's 4Mb in LightiningJS 3.

It would be interesting to be able to measure what is the actual max buffer size we use in an application, in order to minimize this buffer size.

**Comment by marcel-danilewicz-consult-red** [comment]
Hi @elsassph,
I've tried to measure the buffer usage periodically with following:
`gl.getBufferParameter(gl.ARRAY_BUFFER, gl.BUFFER_SIZE)`
but I'm not sure if this is correct. I ran [http://localhost:5173/?test=destroy&multiplier=100](http://localhost:5173/?test=destroy&multiplier=100) and the results are consistent with amount of nodes visible on screen, but for [http://localhost:5173/?test=animation-events](http://localhost:5173/?test=animation-events) the value seems too low to be possible - just 192B.
Could you please give me an advice on that?

**Comment by elsassph** [comment]
I think you should check `curBufferIdx` of the `WebGlCoreRenderer` at the end of rendering a frame, preferably with a sizeable application complexity (hundreds of visual elements).

**Comment by frank-weindel** [comment]
Looks like this issue has evolved into just a way to peek at what the amount of quadBuffer used is?

**Comment by marcel-danilewicz-consult-red** [comment]
Yes, this is how I understood it based on the first comment. Let me know if configuration of quadBuffer size is in the scope of this ticket.

**Comment by elsassph** [comment]
I mean peeking is one thing but we'll eventually want to have an option as well

**Comment by elsassph** [comment]
BTW that's great to contribute @marcel-danilewicz-consult-red but I think it's important to first define well the scope of work, and get a GO from the maintainer.

**Comment by frank-weindel** [comment]
@elsassph Totally agree. Can you make another issue regarding the option? It'll be ripe for a 'good first issue' label.
